packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -459,12 +361,258 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         });
     }
 
-    private async fetchSnapshot(snapshotOptions: string, tokenFetchOptions: TokenFetchOptions) {
+    private isTreesLatestSecondCallEvent(tokenFetchOptions: TokenFetchOptions) {
+        if (tokenFetchOptions.refresh) {
+            // This is the most critical code path for boot.
+            // If we get incorrect / expired token first time, that adds up to latency of boot
+            this.logger.sendErrorEvent({ eventName: "TreeLatest_SecondCall", hasClaims: !!tokenFetchOptions.claims });
+        }
+    }
+
+    private async fetchSnapshotViaPost(
+        tokenFetchOptions: TokenFetchOptions,
```

Comment:
can we add one more query param (say _POst=1) to the body fo the POST request? Service can log that and use to quantify no OPTIONs from our telemetry.

Reply:
Added.